swift/IceDiscovery/helloUI/Client.swift

```diff
@@ -0,0 +1,199 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+import Combine
+import Ice
+import PromiseKit
+
+class Client: ObservableObject {
+    var helloPrx: HelloPrx!
+    var communicator: Ice.Communicator!
+
+    let hostnameKey = "hostnameKey"
+
+    @Published var helloEnabled: Bool = true
+    @Published var shutdownEnabled = true
+    @Published var flushEnabled = false
+    @Published var proxySettings: ProxySettings = .init() {
+        didSet {
+            do { try updateProxy() } catch {
+                exception(error)
+            }
+        }
+    }
+
+    // Error bindings
+    @Published var error: Error?
+    @Published var showingError = false
+
+    // Status bindings
+    @Published var isSpinning = false
+    @Published var statusMessage = ""
+
+    func configure(comm: Ice.Communicator) {
+        communicator = comm
+    }
+
+    func exception(_ err: Error) {
+        error = err
+        showingError = true
+    }
+
+    func sayHello() {
+        do {
+            if helloPrx == nil {
+                return
+            }
+            let delay = Int32(proxySettings.delay)
+            if proxySettings.deliveryMode != .OnewayBatch,
+               proxySettings.deliveryMode != .OnewaySecureBatch,
+               proxySettings.deliveryMode != .DatagramBatch
+            {
+                var response = false
+                firstly {
+                    helloPrx.sayHelloAsync(delay, sentOn: DispatchQueue.main) { _ in
+                        if !response {
+                            self.requestSent()
+                        }
+                    }
+                }.done {
+                    response = true
+                    self.ready()
+                }.catch { error in
+                    response = true
+                    self.exception(error)
+                }
+            } else {
+                try helloPrx.sayHello(delay)
```

Comment:
I think the logic here is it's a batch request, so we "send" it synchronously because we actually only enqueue it locally. Would be nice to add a comment.